### ISSUE TITLE
bug: attribute store ex-request's data

### ISSUE THREAD

**Issue by seansay** [siznax/wptools#23]
now I'm using `wptools.page().get()` to get a page's data, but turn out the data doesn't exist,

but still, the instance it return have these two 
```
{
  g_parse: <dict(3)> {info, query, response}
  g_query: <dict(3)> {info, query, response}
  g_wikidata: <dict(3)> {info, query, response}
  images: <dict(3)> {rimage, rthumb, wimage}
  lang: en
  title: 张楚
  wikidata: <dict(3)> {category, instance, subclass}
}
```

the wikidata is from the ex-request it made,  not the current page

**Comment by siznax** [comment]
Hi, @seansay. This means that the APIs were not able to get any information about your title. If your object is mostly empty after trying a `get`, then you can assume your request found nothing. The tool tries to report missing titles (see below). It is not considered a _fatal_ error, but an operator error. In this case, there is no redirect in English Wikipedia for 张楚 (to Zhang Chu), as you can see here: https://en.wikipedia.org/wiki/%E5%BC%A0%E6%A5%9A

```python
>>> p = wptools.page('张楚').get()
...
en.wikipedia.org (action=query) 张楚
MediaWiki:API missing title: 张楚

en.wikipedia.org (action=parse) 张楚
MediaWiki:API error: {u'info': u"The page you specified doesn't exist", u'code': u'missingtitle', u'docref': u'See https://en.wikipedia.org/w/api.php for API usage'}

en.wikipedia.org (action=wikidata) 张楚
Wikidata missing title: 张楚

张楚 (en)
{
  g_parse: <dict(3)> {info, query, response}
  g_query: <dict(3)> {info, query, response}
  g_wikidata: <dict(3)> {info, query, response}
  lang: en
  title: 张楚
}
```

The `g_` attributes are left in place to allow you to investigate what went wrong. Specifically, you can take a look at what requests were made by loading the `query` in your browser, for example:

```python
>>> p.g_query['query']
'https://en.wikipedia.org/w/api.php?action=query&exintro&format=json&formatversion=2&inprop=displaytitle|url|watchers&list=random&pithumbsize=240&ppprop=wikibase_item&prop=extracts|images|info|pageimages|pageprops&redirects&rnlimit=1&rnnamespace=0&titles=\xe5\xbc\xa0\xe6\xa5\x9a'
```

It's easier to see the API result if you remove `&format=json` from the request:

https://en.wikipedia.org/w/api.php?action=query&exintro&formatversion=2&inprop=displaytitle|url|watchers&list=random&pithumbsize=240&ppprop=wikibase_item&prop=extracts|images|info|pageimages|pageprops&redirects&rnlimit=1&rnnamespace=0&titles=\xe5\xbc\xa0\xe6\xa5\x9a

There you will see:

```
                "missing": true,
```

**Comment by siznax** [comment]
Maybe we should add an "error" attribute when this happens:

```
张楚 (en)
{
  errors: <dict(3)> {get_parse, get_query, get_wikidata}
  ...
}
```

Something like this?

```
>>> p.errors
{'get_parse': "MediaWiki:API error: The page you specified doesn't exist",
 'get_query': 'MediaWiki:API missing title: \xe5\xbc\xa0\xe6\xa5\x9a',
 'get_wikidata': 'Wikidata missing title: \xe5\xbc\xa0\xe6\xa5\x9a'}
```

**Comment by siznax** [comment]
On second thought, API misses should raise `LookupError` early and prevent subsequent requests. I fixed this with 434c365579aac00e0eb2606631964e1679ccb4fa. Thanks for reporting @seansay!
